x-pack/plugins/security_solution/public/common/containers/matrix_histogram/index.test.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { renderHook, act } from '@testing-library/react-hooks';
+
+import { useKibana } from '../../../common/lib/kibana';
+import { useMatrixHistogram } from '.';
+import { MatrixHistogramType } from '../../../../common/search_strategy';
+
+jest.mock('../../../common/lib/kibana');
+
+describe('useMatrixHistogram', () => {
+  const props = {
+    endDate: new Date(Date.now()).toISOString(),
+    errorMessage: '',
+    filterQuery: {},
+    histogramType: MatrixHistogramType.events,
+    indexNames: [],
+    stackByField: 'event.module',
+    startDate: new Date(Date.now()).toISOString(),
+  };
+
+  it('should update request when props has changed', async () => {
+    const { rerender } = renderHook(() => useMatrixHistogram(props));
+
+    props.stackByField = 'event.action';
```

Comment:
given that `props`:

```
  const props = {
    endDate: new Date(Date.now()).toISOString(),
    errorMessage: '',
    filterQuery: {},
    histogramType: MatrixHistogramType.events,
    indexNames: [],
    stackByField: 'event.module',
    startDate: new Date(Date.now()).toISOString(),
  };
```

is shared by both `it` blocks in this `describe`, consider a refactor that prevents this mutation from being seen by the next `it` block